Setup dotenv for environment variables

debelistic opened the pull request:
### What does this PR do?

- Setup dotenv for environment variables
- Add environment variables

### Description of Task to be completed?

- Install ```dotenv``` file
- Add environment variables

### How should this be manually tested?

- Create a file named ```.env``` in root folder and add environment variables
- Import ```config``` from ```dotenv``` to use in any file

### Any background context you want to provide?

- Ensure to ignore the ```.env``` file, to protect secretes

### What are the relevant pivotal tracker stories?

- #167765383

### Implementation Plan

- https://docs.google.com/document/d/1G2GsbzQEKcLkzbvqNjJVPT640QpDynrRVVyay7v-E0s/edit?usp=sharing

### Questions:

debelistic committed:
chore(environment) - setup dotenv

- create .env and .envsample files
- ensure environment variables are added to .env file

[Finishes #167765383]

voke1 commented:
@debelistic Great work. I'd suggest you change `.envSample` to just `.env` as this is required in the root  directory.

debelistic commented:
> @debelistic Great work. I'd suggest you change `.envSample` to just `.env` as this is required in the root directory.

The real `.env` file should be ignored by in gitiginore, so it doesn't show up on github. It contains secrets as well

voke1 commented:
Yes. `.env.` should be gitignored as the same environmental variables will be configured on Travis CI for tests. In this case, I see `.envSample` as being redundant. How do you intend to use `.envSample`?

debelistic commented:
> Yes. `.env.` should be gitignored as the same environmental variables will be configured on Travis CI for tests. In this case, I see `.envSample` as being redundant. How do you intend to use `.envSample`?

The sample file is so you can provide the environment variables without their values. That would give whoever is cloning the repo an idea the environment variables they are meant to set

victor-nach reviewed (commented):
I don't think dotenv should be a dev dependency, you may want to check that

voke1 commented:
> > Yes. `.env.` should be gitignored as the same environmental variables will be configured on Travis CI for tests. In this case, I see `.envSample` as being redundant. How do you intend to use `.envSample`?
> 
> The sample file is so you can provide the environment variables without their values. That would give whoever is cloning the repo an idea the environment variables they are meant to set

Okay. cool. I thought as much. This should be fine

anneKay commented on package.json:
```diff
@@ -26,5 +25,8 @@
     "underscore": "^1.9.1"
+  },
+  "devDependencies": {
+    "dotenv": "^6.2.0"
```
Why did you move 'dotenv' to devdependency?

debelistic replied:
I felt since its only need in development environment, it should be devDependency. But I have moved it to dependencies regardless

debelistic committed:
chore(setup) - setup dotenv

- move module to dependencies
- ignore package-lock.json file

[Finishes #167765383]

debelistic commented:
> I don't think dotenv should be a dev dependency, you may want to check that

I've updated it

ugwumadu116 reviewed (approved):
Nice one. Thanks for your passion

debelistic committed:
chore(setup) - setup dotenv

- add package-lock.json

[Finishes #167765383]

debelistic committed:
Merge branch 'develop' into ch-setup-dotenv-167765383

debelistic committed:
Merge branch 'develop' into ch-setup-dotenv-167765383

anneKay commented on .vscode/settings.json:
```diff
@@ -0,0 +1,5 @@
+  "cSpell.words": [
+    "authorshaven"
+  ]
+}
```
Please gitignore this file

debelistic replied:
> Please gitignore this file

Done

debelistic committed:
chore(environment) - setup dotenv

- ignore unrequired files

[Finishes #167765383]

debelistic committed:
chore(environment) - setup dotenv

- remove lock file conflicts
- ignore unrequired file

[Finishes #167765383]
